src/commands/validate.js

```diff
@@ -40,64 +43,52 @@ function getOverallErrors(variations = [], components = [], log, warn, error) {
 exports.command = 'validate [variations]';
 exports.desc = 'validate Variation Providers';
 exports.builder = (yargs) => {
-  const { project, projects, all } = yargs.argv;
-  const projectCount = projects ? Object.keys(projects).length : 0;
+  const config = normalizeConfig(yargs.argv);
+  const { project, projects, all } = config;
+  const allProjectNames = projects ? Object.keys(projects) : [];
 
-  if (!all) {
-    // this must come after config/pkgConf, so it can be supplied that way.
-    yargs.option('variations', {
-      demandOption: true,
-      requiresArg: true,
-    });
-    yargs.option('components', {
-      demandOption: true,
-      requiresArg: true,
-    });
-  }
-  yargs.positional('variations', {});
-
-  if (all && projectCount <= 0) {
-    throw chalk.red(`\`--all\` requires a non-empty “projects” config`);
+  if (all && allProjectNames.length <= 0) {
```

Comment:
Is it possible for length to be < 0?

Reply:
lol, no, but i prefer this to === 0 for no good reason